mountpoint-s3/src/main.rs

```diff
@@ -309,6 +318,9 @@ fn main() -> anyhow::Result<()> {
                 }
             }
             ForkResult::Parent { child } => {
+                // Re-parse CLI args in parent process because the child might move them
+                let args = CliArgs::parse();
```

Comment:
I was reviewing this and what we already have. The child arm looks a little odd. I can see the child parses again too, but it _also_ uses the original args (in a PR I approved 😄 ).

Should we drop the original args so its clear? At least, I think we should shadow them in the `child` arm.

When reading into fork again, its mapping Copy-on-Write memory for both processes around the original memory of the process before the fork. So re-creating the args is just an ownership thing?

Reply:
Yeah. It should be fine to be COW because it's just plain data. But just in case we ever added something fancier to that struct for some reason, better to just re-parse on both sides.